src/framework/cookies/core.clj

```diff
@@ -5,20 +5,24 @@
     [ring.middleware.cookies :as cookies]
     [xiana.core :as xiana]))
 
-(defn- parse-request-cookies
-  [req]
-  (keywordize-keys
-    (cookies/cookies-request req)))
-
-(defn- parse-response-cookies
-  [resp]
-  (cookies/cookies-response resp))
-
 (def interceptor
   "Parses request and response cookies"
-  {:enter (fn [state]
-            (xiana/ok
-              (update state :request
-                      parse-request-cookies)))
-   :exit  (fn [state]
-            (xiana/ok (update state :response parse-response-cookies)))})
+  (letfn [(move-cookies
```

Comment:
I tried to refactor it a little and ended up with this: 
```
(def interceptor
  "Parses request and response cookies"
  (letfn [(move-cookies [{headers :headers :as req}]
            (cond-> req
              (not (get headers "cookie")) (assoc-in
                                             [:headers "cookie"]
                                             (:cookie headers))))
          (parse-request-cookies [req]
            (-> req move-cookies cookies/cookies-request keywordize-keys))]
    {:enter (fn [state]
              (xiana/ok
                (update state :request
                        parse-request-cookies)))
     :leave (fn [state]
              (xiana/ok (update state :response cookies/cookies-response)))}))
```

Also, why do we need to have a string cookie key? I couldn't found any other usages of it inside framework or Frankie.

Reply:
I like the refactor, seems good. The reason of stringifying the keyword is in ring middlewares:
ring.middleware.cookies/parse-cookies
```clojure
(defn- parse-cookies
  "Parse the cookies from a request map."
  [request encoder]
  (if-let [cookie (get-in request [:headers "cookie"])]
    (->> cookie
         parse-cookie-header
         ((fn [c] (decode-values c encoder)))
         (remove nil?)
         (into {}))
    {}))
```